### ISSUE TITLE
Metal backend with quantized matmul kernels

### ISSUE THREAD

**Issue by anibjoshi** [strata-ai-labs/strata-inference#9]
### Context
strata-core's Metal backend (4,137 LOC) is 95% portable. We port it and add the critical new capability: **fused quantized matmul** — where dequantization happens inside the GPU kernel, never materializing full f32 weight matrices.

### Key New Kernel: Q8_0 Fused Matmul
```metal
// Each block: f16 scale + 32 × i8 values
// Dequant fused with dot product accumulation
kernel void kernel_mul_mv_q8_0_f32(
    device const char * weights_q8,  // quantized weights
    device const float * input,       // f32 input vector
    device float * output,            // f32 output
    ...
) {
    // For each output row:
    //   For each block of 32 values:
    //     scale = block.d
    //     sum += scale * dot(block.qs, input_slice)
    // Uses SIMD group reductions for parallel accumulation
}
```

### Kernel Parameters (from llama.cpp)
```
Q8_0: N_R0=2 rows per SIMD group, N_SG=4 SIMD groups per threadgroup
      → 128 threads total, processes 2 matrix rows in parallel
```

### Additional Kernels Needed
| Kernel | Purpose | Priority |
|--------|---------|----------|
| `kernel_mul_mv_q8_0_f32` | Fused Q8_0 dequant + matmul | P0 |
| `kernel_rms_norm` | RMSNorm for LLaMA/Gemma | P0 |
| `kernel_silu` | SiLU activation | P0 |
| `kernel_swiglu` | SwiGLU gated activation | P0 |
| `kernel_rope` | Rotary position embeddings | P0 |
| `kernel_mul_mv_q4_0_f32` | Fused Q4_0 dequant + matmul | P1 |

### Files to Port from strata-core
- `runtime/metal/mod.rs` (2,094 LOC) → `backend/metal/mod.rs`
- `runtime/metal/ffi.rs` (377 LOC) → `backend/metal/ffi.rs`
- `runtime/metal/kernels.rs` (1,666 LOC) → `backend/metal/kernels.rs` (extend)

### Acceptance Criteria
- [ ] Port all existing Metal kernels (gemm, gelu, layer_norm, softmax, attention_mask, mean_pool, transpose_heads, etc.)
- [ ] New kernel: `kernel_mul_mv_q8_0_f32` with SIMD group reductions
- [ ] New kernel: `kernel_rms_norm` (per-row RMS normalization)
- [ ] New kernel: `kernel_silu` (x * sigmoid(x))
- [ ] New kernel: `kernel_swiglu` (fused silu * up)
- [ ] New kernel: `kernel_rope` (rotary position embeddings)
- [ ] Upload quantized tensors directly to GPU (no f32 conversion)
- [ ] All existing Metal tests pass
- [ ] Benchmark Q8_0 fused matmul vs dequant+matmul (target: >2x speedup)
- [ ] Verify numerical accuracy vs CPU reference (max error < 1e-3)

### Reference
- llama.cpp: `ggml/src/ggml-metal/ggml-metal.metal` lines 3099-3160 (Q4_0/Q8_0 kernels)
- llama.cpp: `ggml/src/ggml-metal/ggml-metal.metal` lines 1760-1850 (softmax kernel)
- strata-core: `runtime/metal/kernels.rs` (existing kernels to port)

### Estimated LOC: ~2,500 (2,000 ported + 500 new kernels)
